### PR TITLE
feat: base `TextField` component

### PR DESCRIPTION
**Pull request by Michael-Liendo**

### CODE REVIEW

File: crates/web/src/components/text_field.rs [hunk 1]
```diff
@@ -82,7 +85,7 @@ pub fn TextField(
     view! {
             <div>
             <label class="block mb-2 text-sm font-medium text-purple-500" for=id.clone()>{label}</label>
-            <input type=format!("{}", r#type) name=name value=value id=id placeholder=placeholder class=class_names  disabled=disabled />
+            <input type=format!("{}", r#type) name=name value=value id=id placeholder=placeholder class=class_names  disabled=disabled on:change=move |ev| {console_log(event_target_value(&ev).as_str())}  />
```
Comment (EstebanBorai) [inline crates/web/src/components/text_field.rs:88]: I think you want to use `on:input` here, based on: https://book.leptos.dev/view/05_forms.html#controlled-inputs

Comment (Michael-Liendo) [reply crates/web/src/components/text_field.rs:88]: Yeah! I am searching how to share the value when the component is used

Comment (EstebanBorai) [reply crates/web/src/components/text_field.rs:88]: I think your best take is to use https://docs.rs/leptos/latest/leptos/callback/index.html.

You could have a prop `on_input` which is a instance of `Callback<String>` and use this `on_input.call(ev.target.value)` there I think.

**Comment by EstebanBorai** [comment]
@Michael-Liendo Im merging this so I con later apply my patch on #210.

**Review by EstebanBorai — APPROVED** [review]

**Pull request merged**